contracts/IdleTokenWrapper.sol

```diff
@@ -0,0 +1,241 @@
+        return _vaultPaused() ? 0 : type(uint256).max;
+    }
+
+    function maxMint(
+        address /* receiver */
+    ) external view returns (uint256) {
+        return _vaultPaused() ? 0 : type(uint256).max;
+    }
+
+    function maxWithdraw(address owner) external view returns (uint256) {
+        return _vaultPaused() ? 0 : convertToAssets(balanceOf(owner));
+    }
+
+    function maxRedeem(address owner) external view returns (uint256) {
+        return _vaultPaused() ? 0 : balanceOf(owner);
+    }
+
+    /// @notice Deposit underlying tokens into IIdleToken
+    /// @dev This function SHOULD be guarded to prevent potential reentrancy
+    /// @param amount Amount of underlying tokens to deposit
+    /// @param receiver receiver of tranche shares
+    /// @param depositor depositor of underlying tokens
+    function _deposit(
+        uint256 amount,
+        address receiver,
+        address depositor
+    ) internal returns (uint256 deposited, uint256 mintedShares) {
+        ERC20Upgradeable _token = ERC20Upgradeable(token);
+
+        SafeERC20Upgradeable.safeTransferFrom(_token, depositor, address(this), amount);
+
+        uint256 beforeBal = _token.balanceOf(address(this));
+        mintedShares = idleToken.mintIdleToken(amount, true, address(this));
+        uint256 afterBal = _token.balanceOf(address(this));
+
+        deposited = beforeBal - afterBal;
+
+        _mint(receiver, mintedShares);
+    }
+
+    /// @notice Withdraw underlying tokens from IIdleToken
+    /// @dev This function SHOULD be guarded to prevent potential reentrancy
+    /// @param shares shares to withdraw
+    /// @param receiver receiver of underlying tokens withdrawn from IIdleToken
+    /// @param sender sender of tranche shares
+    function _redeem(
+        uint256 shares,
+        address receiver,
+        address sender
+    ) internal returns (uint256 withdrawn, uint256 burntShares) {
+        IIdleToken _idleToken = idleToken;
+
+        // withdraw from idleToken
+        uint256 beforeBal = _idleToken.balanceOf(address(this));
+        withdrawn = _idleToken.redeemIdleToken(shares);
+        burntShares = beforeBal - _idleToken.balanceOf(address(this));
+
+        _burnFrom(sender, burntShares);
+        SafeERC20Upgradeable.safeTransfer(ERC20Upgradeable(token), receiver, withdrawn);
+    }
+
+    function _burnFrom(address account, uint256 amount) internal {
+        if (account != msg.sender) {
+            uint256 currentAllowance = allowance(account, msg.sender);
+            require(currentAllowance >= amount, "idleTokenWrapper: burn amount exceeds allowance");
+            unchecked {
+                _approve(account, msg.sender, currentAllowance - amount);
+            }
+        }
+        _burn(account, amount);
+    }
+
+    function _vaultPaused() internal view returns (bool) {
+        (bool success, bytes memory data) = address(idleToken).staticcall(abi.encodeWithSignature("paused()"));
```

Comment:
Why using `staticcall` insteal of `idleToken.paused()` ?

Reply:
`IIdleToken` interface don't inherit `Pausable` . 
maybe we can change `idleToken` type to `IdleTokenFungible` type.

Reply:
I think we can use a custom IdleToken interface then instead were we add the paused method